### PR TITLE
chore: simplify datapipe and test orphan file cleanup

### PR DESCRIPTION
**Pull request by zinic**
## Description

This is a continuation of work done earlier. Mostly clean up and a small refactor to pull additional complexity from the datapipe and add test coverage.

## Motivation and Context

Further clean up from earlier datapipe refactors.

## How Has This Been Tested?

Added additional unit tests to cover this refactor.

## Types of changes

-   [x] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/daemons/datapipe/datapipe.go [hunk 1]
```diff
@@ -190,44 +187,15 @@ func (s *Daemon) Stop(ctx context.Context) error {
...
-	defer s.clearOrphanedFilesLock.Unlock()
-
-	relativeTmpDir := s.cfg.TempDirectory()
-
-	if orphanFiles, err := os.ReadDir(s.cfg.TempDirectory()); err != nil {
-		log.Errorf("Failed fetching available files: %v", err)
-	} else if ingestTasks, err := s.db.GetAllIngestTasks(); err != nil {
-		log.Errorf("Failed fetching available ingest tasks: %v", err)
+	if ingestTasks, err := s.db.GetAllIngestTasks(); err != nil {
+		log.Errorf("")
```
Comment (computator) [inline cmd/api/src/daemons/datapipe/datapipe.go:191]: Is this missing error message intentional?

Comment (zinic) [reply cmd/api/src/daemons/datapipe/datapipe.go:191]: NOPE

One moment.

Comment (zinic) [reply cmd/api/src/daemons/datapipe/datapipe.go]: Patched up

**Review by rvazarkar — APPROVED** [review]
Looks good, and extra testing is always great. A comment explaining how the ExpectedFile test works would be great as its not immediately obvious (at least to me)

**Pull request merged**